examples/morpheusvm/controller/controller.go

```diff
@@ -18,24 +15,17 @@ import (
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/consts"
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/genesis"
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/registry"
-	"github.com/ava-labs/hypersdk/examples/morpheusvm/rpc"
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
-	"github.com/ava-labs/hypersdk/extension/indexer"
-	"github.com/ava-labs/hypersdk/pebble"
 	"github.com/ava-labs/hypersdk/vm"
-
-	ametrics "github.com/ava-labs/avalanchego/api/metrics"
-	hrpc "github.com/ava-labs/hypersdk/rpc"
-	hstorage "github.com/ava-labs/hypersdk/storage"
 )
 
 var (
-	_ vm.Controller        = (*Controller)(nil)
-	_ vm.ControllerFactory = (*factory)(nil)
+	_ vm.Controller                     = (*Controller)(nil)
+	_ vm.ControllerFactory[*Controller] = (*factory)(nil)
 )
 
-func New(options ...vm.Option) (*vm.VM, error) {
-	return vm.New(
+func New(options ...vm.Option[*Controller]) (*vm.VM[*Controller], error) {
```

Comment:
The controller should define the default options ie. supported APIs, subscriptions, etc. here or even further down (set as defaults in the VM itself) rather than pushing it to the caller of this function to decide which APIs to enable